sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -283,7 +284,7 @@ class HadoopTableReader(
       _minSplitsPerRDD)
 
     // Only take the value (skip the key) because Hive works only with values.
-    rdd.map(_._2)
+    CombineSmallFile.combineWithPath(rdd, sc, path).map(_._2)
```

Comment:
The problem is that `CombineSmallFile.combineWithPath()` always lists files under `path` no matter combining is enabled or not.